Backpack/Calendar/Classes/BPKCalendar.m

```diff
@@ -755,6 +790,64 @@ + (BOOL)date:(NSDate *)date isBetweenDate:(NSDate *)beginDate andDate:(NSDate *)
     return YES;
 }
 
+- (RowType)rowTypeForDate:(NSDate *)date amongRangeOfDates:(NSArray<NSDate *> *)datesRange {
+    if ([self isDate:date betweenRangeOfDates:datesRange]) {
+        BOOL isRowStart = [self isFirstDayOfTheWeekOrTheMonth:date];
+        BOOL isRowEnd = [self isLastDayOfTheWeekOrTheMonth:date];
+
+        if (isRowStart && isRowEnd) {
+            return RowTypeBoth;
+        } else if (isRowStart) {
+            return RowTypeStart;
+        } else if (isRowEnd) {
+            return RowTypeEnd;
+        } else {
+            return RowTypeMiddle;
+        }
+    } else {
+        return RowTypeMiddle;
+    }
+}
+
+- (SelectionType)selectionTypeForDate:(NSDate *)date amongRangeOfDates:(NSArray<NSDate *> *)datesRange {
+    if ([self isDate:date betweenRangeOfDates:datesRange]) {
+        NSDate *minDate = [datesRange firstObject];
+        NSDate *maxDate = [datesRange lastObject];
+        BOOL isMinDate = [date isEqualToDate:minDate];
+        BOOL isMaxDate = [date isEqualToDate:maxDate];
+
+        if (isMinDate) {
+            return SelectionTypeLeadingBorder;
+        } else if (isMaxDate) {
+            return SelectionTypeTrailingBorder;
+        } else {
+            return SelectionTypeMiddle;
+        }
+    } else {
+        return SelectionTypeNone;
+    }
+}
+
+- (BOOL)isDate:(NSDate *)date betweenRangeOfDates:(NSArray<NSDate *> *)dates {
+    NSDate *minDate = [dates firstObject];
+    NSDate *maxDate = [dates lastObject];
+    return [BPKCalendar date:date isBetweenDate:minDate andDate:maxDate];
+}
+
+- (BOOL)isFirstDayOfTheWeekOrTheMonth:(NSDate *)date {
+    NSCalendar *gregorian = self.calendarView.gregorian;
+    NSDate *firstWeekday = [gregorian fs_firstDayOfWeek:date];
+    NSDate *firstDayOfMonth = [gregorian fs_firstDayOfMonth:date];
+    return [gregorian isDate:date inSameDayAsDate:firstWeekday] || [gregorian isDate:date inSameDayAsDate:firstDayOfMonth];
+}
+
+- (BOOL)isLastDayOfTheWeekOrTheMonth:(NSDate *)date {
+    NSCalendar *gregorian = self.calendarView.gregorian;
+    NSDate *lastWeekday = [gregorian fs_lastDayOfWeek:date];
+    NSDate *lastDayOfMonth = [gregorian fs_lastDayOfMonth:date];
+    return [gregorian isDate:date inSameDayAsDate:lastWeekday] || [gregorian isDate:date inSameDayAsDate:lastDayOfMonth];
+}
```

Comment:
Is this the right location for these helper functions? Should we 'isolate' them to mark them as only in use for the wholeMonth selection?

Reply:
This helper methods are product of refactoring `- (void)configureCell:(FSCalendarCell *)cell forDate:(NSDate *)date atMonthPosition:(FSCalendarMonthPosition)monthPosition`.

I think right now is more readable and understandable than before, even if we get rid of the whole month selection.

We have passed from:
```
SelectionType selectionType = SelectionTypeNone;
RowType rowType = RowTypeMiddle;

if (!self.sameDayRange && sortedSelectedDates.count > 1 && self.selectionConfiguration.isRangeStyleSelection) {
            NSDate *minDate = [sortedSelectedDates firstObject];
            NSDate *maxDate = [sortedSelectedDates lastObject];
            BOOL dateInsideRange = [BPKCalendar date:date isBetweenDate:minDate andDate:maxDate];
            if (dateInsideRange) {
                BOOL isMinDate = [date isEqualToDate:minDate];
                BOOL isMaxDate = [date isEqualToDate:maxDate];
                NSCalendar *gregorian = self.calendarView.gregorian;
                NSDate *firstWeekday = [gregorian fs_firstDayOfWeek:date];
                NSDate *lastWeekday = [gregorian fs_lastDayOfWeek:date];
                NSDate *firstDayOfMonth = [gregorian fs_firstDayOfMonth:date];
                NSDate *lastDayOfMonth = [gregorian fs_lastDayOfMonth:date];
                BOOL isRowStart = [gregorian isDate:date inSameDayAsDate:firstWeekday] || [gregorian isDate:date inSameDayAsDate:firstDayOfMonth];
                BOOL isRowEnd = [gregorian isDate:date inSameDayAsDate:lastWeekday] || [gregorian isDate:date inSameDayAsDate:lastDayOfMonth];

                if (isRowStart && isRowEnd) {
                    rowType = RowTypeBoth;
                } else if (isRowStart) {
                    rowType = RowTypeStart;
                } else if (isRowEnd) {
                    rowType = RowTypeEnd;
                }

                if (isMinDate) {
                    selectionType = SelectionTypeLeadingBorder;
                } else if (isMaxDate) {
                    selectionType = SelectionTypeTrailingBorder;
                } else {
                    selectionType = SelectionTypeMiddle;
                }
            }
        }
}
```
To this:
```
SelectionType selectionType = SelectionTypeNone;
RowType rowType = RowTypeMiddle;
BOOL wholeMonthSelection = NO;

if (self.isSelectingWholeMonth) {
            rowType = [self rowTypeForDate:date amongRangeOfDates:sortedSelectedDates];
            selectionType = [self selectionTypeForDate:date amongRangeOfDates:sortedSelectedDates];
            wholeMonthSelection = [self isDate:date betweenRangeOfDates:sortedSelectedDates];
}
```

Reply:
Oh wow, that's a massive improvement! Thank you :D